airflow/models.py

```diff
@@ -1458,6 +1527,9 @@ def signal_handler(signum, frame):
                     raise AirflowException("Task received SIGTERM signal")
                 signal.signal(signal.SIGTERM, signal_handler)
 
+                # Give operators a chance to save xcom data from previous runs.
+                task_copy.save_previous_xcoms(context=context)
+
```

Comment:
New hook in TaskInstance!  (You can implement it in your operator - airflow inheritance is weird)